txpipe/twopoint.py

```diff
@@ -385,299 +362,171 @@ def call_treecorr(self, data, meta, i, j, k):
-        f.close()
-
-        f = self.open_input('lens_tomography_catalog')
-        lens_bin = f['tomography/lens_bin'][:]
-        f.close()
-
-        data['source_bin']  =  source_bin
-        data['lens_bin']  =  lens_bin
-        data['R']  =  r_total
-
-    def load_lens_catalog(self, data):
-        # Subclasses can load an external lens catalog
-        pass
-
-
+            cat_j = self.get_lens_catalog(j)
+            rancat_j = self.get_random_catalog(j)
 
-    def load_shear_catalog(self, data):
-
-        # Columns we need from the shear catalog
-        read_shear_catalog_type(self)
-
-        if self.config['shear_catalog_type']=='metacal':
-            if self.config['use_true_shear']:
-                cat_cols = ['ra', 'dec', 'true_g1', 'true_g2', 'mcal_flags']
-            else:
-                cat_cols = ['ra', 'dec', 'mcal_g1', 'mcal_g2', 'mcal_flags']
+        t1 = perf_counter()
+        
+        nn = treecorr.NNCorrelation(self.config)
+        nn.process(cat_i, cat_j, comm=self.comm)
+        
+        nr = treecorr.NNCorrelation(self.config)
+        nr.process(cat_i, rancat_j, comm=self.comm)
 
+        # The next calculation is faster if we explicitly tell TreeCorr
+        # that its two catalogs here are the same one.
+        if i == j:
+            rancat_j = None
+        
+        rr = treecorr.NNCorrelation(self.config)
+        rr.process(rancat_i, rancat_j, comm=self.comm)
+        
+        if i==j:
+            rn = None
         else:
-            cat_cols = ['ra', 'dec', 'g1', 'g2', 'weight','flags','sigma_e','m']
-        print(f"Loading shear catalog columns: {cat_cols}")
+            rn = treecorr.NNCorrelation(self.config)
+            rn.process(rancat_i, cat_j, comm=self.comm)
 
-        f = self.open_input('shear_catalog')
-        g = f['shear']
-        for col in cat_cols:
-            print(f"Loading {col}")
-            data[col] = g[col][:]
-
-
-    def load_random_catalog(self, data):
-        filename = self.get_input('random_cats')
-        if filename is None:
-            print("Not using randoms")
-            return
-
-        # Columns we need from the tomography catalog
-        randoms_cols = ['dec','ra','bin']
-        print(f"Loading random catalog columns: {randoms_cols}")
+        if self.rank == 0:
+            t2 = perf_counter()
+            nn.calculateXi(rr, dr=nr, rd=rn)
+            print(f"Processing took {t2 - t1:.1f} seconds")            
 
-        f = self.open_input('random_cats')
-        group = f['randoms']
+        return nn
 
-        cut = self.config['reduce_randoms_size']
-        if 0.0<cut<1.0:
-            N = group['dec'].size
-            sel = np.random.uniform(size=N) < cut
-        else:
-            sel = slice(None)
 
-        data['random_ra'] =  group['ra'][sel]
-        data['random_dec'] = group['dec'][sel]
-        data['random_bin'] = group['bin'][sel]
 
-        f.close()
 
     def calculate_area(self, data):
```

Comment:
I know this is not part of the review, but this seems a little dangerous in that 1) nside is fixed 2) for galaxy density, sometimes the weights should be taken into account... maybe at least put a comment for future improvement? also just searching through, is this actually used?

Reply:
Thanks - this isn't used any more so I've removed.  But your questions apply to the place where area is calculated, and I've added a TODO there.